### PR TITLE
feat: pub prometheus sink configs

### PR DESCRIPTION
**Pull request by suikammd**
close issue #19517

### CODE REVIEW

**Comment by StephenWakely** [comment]
@suikammd I think CI is exposing further errors to the rust-doc generation due to making these fields public. Would you be able to `cd rust-doc; make docs` and have a look at the errors it is now raising? It's unlikely to be caused by the changes you are making, so if you have any questions on how to fix them let me know.

**Comment by suikammd** [comment]
@StephenWakely I find we should use vector_core in dev dependencies. After add this dependency, `make docs` passed.

You may check the new commit. thx!

**Comment by suikammd** [comment]
@StephenWakely  I find we should add vector_core in dependencies. After add this dependency, `make docs` passed.
It is wired that add dependency to dependencies since we do not need it to build vector crate. I tried add this dependency to dev-dependencies, it doesn't work like described in this [link](https://github.com/rust-lang/cargo/issues/3475).

You may check the new commit. thx!

**Review by pront — APPROVED** [review]

**Review by StephenWakely — CHANGES_REQUESTED** [review]
@suikammd The error is because those symbols are being imported into Vector via `vector_lib::sink::VectorSink`. So rather changing Vector to import `vector_core`, can you update the links to point to `vector_lib`. You could probably remove the path altogether since the compiler should know where to find it:

```
//! Contains the [`VectorSink`] instance that is responsible
```

If you feel inclined could you change this comment:

```
//! The Prometheus Remote Write [`vector_core::sink::VectorSink`]
```

to just say


```
//! The Prometheus Remote Write sink
```

It doesn't really make sense to link to the trait in the comment there anyway.

File: src/sinks/prometheus/remote_write/mod.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
-//! The Prometheus Remote Write [`vector_core::sink::VectorSink`].
-//! Contains the [`vector_core::sink::VectorSink`] instance that is responsible
+//! The Prometheus Remote Write.
```
Comment (StephenWakely) [inline src/sinks/prometheus/remote_write/mod.rs:1]:
```suggestion
//! The Prometheus Remote Write sink.
```

File: src/sinks/prometheus/remote_write/mod.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
+//! Contains the [`VectorSink`] instance that is responsible
 //! for taking a stream of [`vector_core::event::Event`] and forwarding
```
Comment (StephenWakely) [inline src/sinks/prometheus/remote_write/mod.rs:3]:
```suggestion
//! for taking a stream of [`Event`] and forwarding
```

**Review by StephenWakely — CHANGES_REQUESTED** [review]

**Review by StephenWakely — APPROVED** [review]

**Pull request merged**